test/framework/deprecated.go

```diff
@@ -0,0 +1,270 @@
+	// fail.
+	By("creating a Cluster resource linked to the InfrastructureCluster resource")
+	Eventually(func() error {
+		if err := mgmtClient.Create(ctx, input.Cluster); err != nil {
+			fmt.Printf("%+v\n", err)
+			return err
+		}
+		return nil
+	}, input.CreateTimeout, eventuallyInterval).Should(BeNil())
+
+	By("creating related resources")
+	for i := range input.RelatedResources {
+		obj := input.RelatedResources[i]
+		By(fmt.Sprintf("creating a/an %s resource", obj.GetObjectKind().GroupVersionKind()))
+		Eventually(func() error {
+			return mgmtClient.Create(ctx, obj)
+		}, input.CreateTimeout, eventuallyInterval).Should(BeNil())
+	}
+
+	By("creating the machine template")
+	Expect(mgmtClient.Create(ctx, input.MachineTemplate)).To(Succeed())
+
+	By("creating a KubeadmControlPlane")
+	Eventually(func() error {
+		err := mgmtClient.Create(ctx, input.ControlPlane)
+		if err != nil {
+			fmt.Println(err)
+		}
+		return err
+	}, input.CreateTimeout, 10*time.Second).Should(BeNil())
+
+	By("waiting for cluster to enter the provisioned phase")
+	Eventually(func() (string, error) {
+		cluster := &clusterv1.Cluster{}
+		key := client.ObjectKey{
+			Namespace: input.Cluster.GetNamespace(),
+			Name:      input.Cluster.GetName(),
+		}
+		if err := mgmtClient.Get(ctx, key, cluster); err != nil {
+			return "", err
+		}
+		return cluster.Status.Phase, nil
+	}, input.CreateTimeout, eventuallyInterval).Should(Equal(string(clusterv1.ClusterPhaseProvisioned)))
+
+	// Create the machine deployment if the replica count >0.
+	if machineDeployment := input.MachineDeployment.MachineDeployment; machineDeployment != nil {
+		if replicas := machineDeployment.Spec.Replicas; replicas != nil && *replicas > 0 {
+			By("creating a core MachineDeployment resource")
+			Expect(mgmtClient.Create(ctx, machineDeployment)).To(Succeed())
+
+			By("creating a BootstrapConfigTemplate resource")
+			Expect(mgmtClient.Create(ctx, input.MachineDeployment.BootstrapConfigTemplate)).To(Succeed())
+
+			By("creating an InfrastructureMachineTemplate resource")
+			Expect(mgmtClient.Create(ctx, input.MachineDeployment.InfraMachineTemplate)).To(Succeed())
+		}
+
+		By("waiting for the workload nodes to exist")
+		Eventually(func() ([]v1.Node, error) {
+			workloadClient, err := input.Management.GetWorkloadClient(ctx, input.Cluster.Namespace, input.Cluster.Name)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get workload client")
+			}
+			nodeList := v1.NodeList{}
+			if err := workloadClient.List(ctx, &nodeList); err != nil {
+				return nil, err
+			}
+			return nodeList.Items, nil
+		}, input.CreateTimeout, 10*time.Second).Should(HaveLen(int(*machineDeployment.Spec.Replicas)))
+	}
+
+	By("waiting for all machines to be running")
+	inClustersNamespaceListOption := client.InNamespace(input.Cluster.Namespace)
+	matchClusterListOption := client.MatchingLabels{clusterv1.ClusterLabelName: input.Cluster.Name}
+	Eventually(func() (bool, error) {
```

Comment:
Here and there, it might be worth while to consider using [wait.Poll](https://godoc.org/k8s.io/apimachinery/pkg/util/wait), and definitely worthwhile to add a [retryable test like this](https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/utils/create_resources.go#L55)

Reply:
Eventually will do the polling for us.

I also don't want to change signatures or anything about the functions in deprecated.go as they exist solely not to break existing consumers.